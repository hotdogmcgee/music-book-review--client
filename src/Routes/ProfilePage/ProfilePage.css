.profile-stats {
  display: flex;
  flex-direction: column;
  align-items: center;
}
