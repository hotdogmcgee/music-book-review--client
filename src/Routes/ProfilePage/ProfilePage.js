import React from "react";
import { Section, NiceDate } from "../../Components/Utils/Utils";
import TokenService from "../../services/token-service";
import BooksApiService from "../../services/books-api-service";
import BookListContext from "../../Contexts/BookListContext";
import MyReviewsList from "../../Components/MyReviewsList/MyReviewsList.js";
import "./ProfilePage.css";

export default class ProfilePage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      reviews: [],
      reviewsOrSaved: null
    };
  }

  static contextType = BookListContext;

  componentDidMount() {
    BooksApiService.getAllUserReviews().then(reviews =>
      this.setReviews(reviews)
    );
  }

  setReviews(reviews) {
    this.setState({ reviews });
  }

  toggleReviewsOrSavedBooks = value => {
    this.setState({ reviewsOrSaved: value });
  };

  renderProfile = () => {
    const dateJoined = TokenService.getUserDateCreated();
    return (
      <section>
        <div className="profile-stats">
          <p>
            Date Joined: <NiceDate date={dateJoined} />
          </p>
          <p>Number of Reviews: {this.state.reviews.length || 0}</p>
        </div>
      </section>
    );
  };
  render() {
    return (
      <Section id="profile-page">
        {this.renderProfile()}
        <MyReviewsList
          reviews={this.state.reviews}
          bookList={this.context.bookList}
        />
      </Section>
    );
  }
}
