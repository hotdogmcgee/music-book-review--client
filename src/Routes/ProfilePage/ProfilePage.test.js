import React from "react";
import ProfilePage from "./ProfilePage";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("ProfilePage component", () => {
  it("renders a ProfilePage by default", () => {
    const wrapper = shallow(<ProfilePage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});