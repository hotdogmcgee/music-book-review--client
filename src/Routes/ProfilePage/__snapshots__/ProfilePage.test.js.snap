// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage component renders a ProfilePage by default 1`] = `
<Section
  id="profile-page"
>
  <section>
    <div
      className="profile-stats"
    >
      <p>
        Date Joined: 
        <NiceDate
          date={null}
        />
      </p>
      <p>
        Number of Reviews: 
        0
      </p>
    </div>
  </section>
  <MyReviewsList
    reviews={Array []}
  />
</Section>
`;
